### ISSUE TITLE
Fluxgym Failure?

### ISSUE THREAD

**Issue by Metairieman55** [cocktailpeanut/fluxgym#238]
I'm only about 6 months in the AI image inference game. I've been experimenting with ComfyUI, Forge, KoyhaSS and just recently I'm playing around with Pinokio specifically Fluxgym. I like all the choices to experiment with so kudos to that. 

I recently decided to expand my AI world and bought a couple of dedicated drives, one especially for Pinokio. I had been sharing drives and running out of space so . . . I had tried a couple of Fluxgym trainings and just last night I tried virtually the same subject on this new setup. And I'm getting a recurring problem just like the older setup. What I'm training spits out samples that have nothing to do with the subject I intended. In other words, I am using a set of portrait snapshots (21) from one specific woman trying to make a face focus Lora to use as a consistent character project. That was/is my goal. However, all I'm getting is completely unrelated returns in the sampling steps.

Here's a sample of a prompt I used in the Florence-2 process: adg4l a girl in a green dress and sunglasses smiling at the camera. She is wearing a necklace and earrings.

Nothing elaborate, same girl in every picture in different clothes, stances, etc., no NSFW stuff just portraits. Below are the results:
A Black person in a non-focused mall(?)
![adface_000050_00_20241105232450](https://github.com/user-attachments/assets/40ae9a12-c01d-4b56-9033-cede7257e09a)
Two people with long hair in a stadium
![adface_000100_00_20241105233539](https://github.com/user-attachments/assets/44eeaf88-1f06-402d-99cc-3cdf27f84e78)
A neon Chinese sign hanging over a street scene
![adface_000150_00_20241105234627](https://github.com/user-attachments/assets/566962b6-3b3c-4636-9ec9-00386df50c5d)

Here are the training parameters:

accelerate launch ^
  --mixed_precision bf16 ^
  --num_cpu_threads_per_process 1 ^
  sd-scripts/flux_train_network.py ^
  --pretrained_model_name_or_path "P:\Pinokio\api\fluxgym.git\models\unet\bdsqlsz\flux1-dev2pro-single\flux1-dev2pro.safetensors" ^
  --clip_l "P:\Pinokio\api\fluxgym.git\models\clip\clip_l.safetensors" ^
  --t5xxl "P:\Pinokio\api\fluxgym.git\models\clip\t5xxl_fp16.safetensors" ^
  --ae "P:\Pinokio\api\fluxgym.git\models\vae\ae.sft" ^
  --cache_latents_to_disk ^
  --save_model_as safetensors ^
  --sdpa --persistent_data_loader_workers ^
  --max_data_loader_n_workers 2 ^
  --seed 42 ^
  --gradient_checkpointing ^
  --mixed_precision bf16 ^
  --save_precision bf16 ^
  --network_module networks.lora_flux ^
  --network_dim 4 ^
  --optimizer_type adafactor ^
  --optimizer_args "relative_step=False" "scale_parameter=False" "warmup_init=False" ^
  --lr_scheduler constant_with_warmup ^
  --max_grad_norm 0.0 ^--sample_prompts="P:\Pinokio\api\fluxgym.git\outputs\adface\sample_prompts.txt" --sample_every_n_steps="50" ^
  --learning_rate 8e-4 ^
  --cache_text_encoder_outputs ^
  --cache_text_encoder_outputs_to_disk ^
  --fp8_base ^
  --highvram ^
  --max_train_epochs 10 ^
  --save_every_n_epochs 1 ^
  --dataset_config "P:\Pinokio\api\fluxgym.git\outputs\adface\dataset.toml" ^
  --output_dir "P:\Pinokio\api\fluxgym.git\outputs\adface" ^
  --output_name adface ^
  --timestep_sampling shift ^
  --discrete_flow_shift 3.1582 ^
  --model_prediction_type raw ^
  --guidance_scale 1 ^
  --loss_type l2 ^

What am I misunderstanding? I am running an RTX4060TI with 16GB VRAM set on 16GB. I'm putting in apples but getting avocados & oranges so to speak . . . Any insights would be much appreciated! Thanks!

**Comment by TimRutjens04** [comment]
What prompts are you using to generate your samples? Have you tried changing your learning rate (maybe test something like 1e-3)? Maybe also change your network dimension to something higher like 8 or 16(--network_dim 8).

**Comment by Metairieman55** [comment]
> Were you consistent in using "adg4l a girl in a green dress" for all your captions? (Or at least using "adg4l" in all of them?) AFAIK, you have to use that same term in your sample prompts to expect a trained result.

At my level of knowledge right now, I did use the trigger word in inference. Now the "sample prompts" I have seen but never knew what that meant! I'll try that to see what I get . . . Thanks.

>What prompts are you using to generate your samples? Have you tried changing your learning rate (maybe test something like 1e-3)? Maybe also change your network dimension to something higher like 8 or 16(--network_dim 8).

At this point I don't remember. I think I just used the defaults (I was following video instructions) to see what would happen. I do seem to remember pumping up the network dimension maybe to 128 or 256 as that was suggested.

Right now, I'm going to retry using both your suggestions. Thanks for the replies!
